infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -275,8 +285,10 @@ public void updateReplicationMetadata(TxnContext txn, LogReplicationSession sess
      * @param incoming              true, if session is incoming (sink), false otherwise (source)
      */
     public void addSession(TxnContext txn, LogReplicationSession session, long topologyConfigId, boolean incoming) {
-        log.info("Add entry to metadata manager, session={}, config_id={}, incoming={}", session, topologyConfigId, incoming);
-        initializeMetadata(txn, session, incoming, topologyConfigId);
+        if(isLeader.get()) {
```

Comment:
what if a leader has not been elected when this method is invoked?

Reply:
sessionManager.refresh() is called everytime leadership is acquired...so leader, even if it is after a delay, will write an initial value for a new session, and only then will the replication events start.